packages/labs/analyzer/src/lib/analyzer.ts

```diff
@@ -29,15 +32,19 @@ export class Analyzer implements AnalyzerInterface {
   // or any of its dependencies change
   readonly moduleCache = new Map<AbsolutePath, Module>();
   private readonly _getProgram: () => ts.Program;
+  readonly typescript: TypeScript;
   readonly fs: AnalyzerInterface['fs'];
   readonly path: AnalyzerInterface['path'];
   private _commandLine: ts.ParsedCommandLine | undefined = undefined;
   private readonly diagnostics: ts.Diagnostic[] = [];
 
   constructor(init: AnalyzerInit) {
-    this._getProgram = init.getProgram;
-    this.fs = init.fs;
-    this.path = init.path;
+    ({
+      fs: this.fs,
+      path: this.path,
+      typescript: this.typescript,
+      getProgram: this._getProgram,
+    } = init);
```

Comment:
Is this syntax any better? a sequence of assignments seems more legible and more likely to be optimized by VMs

Reply:
In size constrained libraries it removes a dereference per property, and I've started to like the look of the pattern. It's also nice for code-completion when you add a new option to the init interface.